### PR TITLE
BOM-2040 : Post Python 3.8 changes

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Relevant JIRA : https://openedx.atlassian.net/browse/BOM-2040

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

**Review by Kelketek — APPROVED** [review]
@mzulqarnain1 I'm glad to see these changes. I looked over BOM-2040 and noticed a couple of changes not mentioned there but which should be done as well (looks like the ticket was created before we actually did the switch). I will create a follow-up PR for them. This PR looks good from my perspective, however.

Ran the integration tests as well and they seem to be passing.

cc @bradenmacdonald

File: blockstore/apps/api/v1/tests/test_contract.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def test_basic_draft_commit(self):
         assert staged_draft_data['base_snapshot'] is None  # No commit yet.
 
         # Bundle should show our draft, but no versions (no commits yet)
-        bundle_detail_response = self.client.get(u'/api/v1/bundles/{}'.format(draft_data["bundle_uuid"]))
+        bundle_detail_response = self.client.get('/api/v1/bundles/{}'.format(draft_data["bundle_uuid"]))
```
[the remark this answers is not in the document]
Comment (mzulqarnain1) [reply blockstore/apps/api/v1/tests/test_contract.py:289]: I used `pyupgrade` package to do this. That seems to miss these for some reason, I'll do these manually.

File: blockstore/apps/api/v1/tests/test_contract.py [hunk 2]
```diff
@@ -169,14 +169,14 @@ def test_create(self):
...
-        assert create_data['collection'] == u'http://testserver/api/v1/collections/{}'.format(self.collection_uuid_str)
+        assert create_data['collection'] == f'http://testserver/api/v1/collections/{self.collection_uuid_str}'
         assert create_data['collection_uuid'] == self.collection_uuid_str
         assert create_data['description'] == "This is a 😀😀😀😀 Bundle"
         assert create_data['drafts'] == {}
         assert create_data['slug'] == 'Ηαρργ'
         assert create_data['title'] == "Happy Bundle 😀"
         assert re.match(UUID4_REGEX, create_data['uuid'])
-        assert create_data['url'] == u"http://testserver/api/v1/bundles/{}".format(create_data['uuid'])
+        assert create_data['url'] == "http://testserver/api/v1/bundles/{}".format(create_data['uuid'])
```
[the remark this answers is not in the document]
Comment (mzulqarnain1) [reply blockstore/apps/api/v1/tests/test_contract.py]: Seems like `pyupgrade` package ignores such strings, I've done these manually.

**Comment by mzulqarnain1** [comment]
> @mzulqarnain1 I'm glad to see these changes. I looked over BOM-2040 and noticed a couple of changes not mentioned there but which should be done as well (looks like the ticket was created before we actually did the switch). I will create a follow-up PR for them. This PR looks good from my perspective, however.
> 
> Ran the integration tests as well and they seem to be passing.
> 
> cc @bradenmacdonald

@Kelketek Thanks. I've made some more fstrings related changes. Please let me know when it's safe to merge this PR.

**Comment by mzulqarnain1** [comment]
@Kelketek I see that https://github.com/edx/blockstore/pull/91 is merge, which had these changes included. Should I go on and close this PR?

**Comment by Kelketek** [comment]
@mzulqarnain1 Rebase-- it was merged before your last formatting changes.

**Pull request merged**